dl4j-cv-labs/src/main/java/global/skymind/solution/object_detection/MetalDefectsDetector/MetalSurfaceDefectDetector_YOLOv2.java

```diff
@@ -0,0 +1,236 @@
+import org.deeplearning4j.nn.weights.WeightInit;
+import org.deeplearning4j.optimize.listeners.ScoreIterationListener;
+import org.deeplearning4j.ui.api.UIServer;
+import org.deeplearning4j.ui.stats.StatsListener;
+import org.deeplearning4j.ui.storage.InMemoryStatsStorage;
+import org.deeplearning4j.util.ModelSerializer;
+import org.deeplearning4j.zoo.model.YOLO2;
+import org.nd4j.linalg.activations.Activation;
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.dataset.api.preprocessor.ImagePreProcessingScaler;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.learning.config.Adam;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.awt.event.KeyEvent;
+import java.io.File;
+import java.util.List;
+
+import static org.bytedeco.opencv.global.opencv_core.CV_8U;
+import static org.bytedeco.opencv.global.opencv_core.flip;
+import static org.bytedeco.opencv.global.opencv_imgproc.*;
+import static org.bytedeco.opencv.helper.opencv_core.RGB;
+
+///**
+// * This is an example of a metal surface defect detection using YOLOv2 architecture.
+// * If no model exists, train a model using Transfer Learning, then validate with test set
+// * If model exists, Validate model with test set.
+// * Data Source: http://faculty.neu.edu.cn/yunhyan/NEU_surface_defect_database.html
+// * **/
+
+public class MetalSurfaceDefectDetector_YOLOv2 {
+    private static final Logger log = LoggerFactory.getLogger(MetalSurfaceDefectDetector_YOLOv2.class);
+    private static int seed = 123;
+    private static double detectionThreshold = 0.5;
+    private static int nBoxes = 5;
+    private static double lambdaNoObj = 0.5;
+    private static double lambdaCoord = 5.0;
+    private static double[][] priorBoxes = {{1, 4}, {2.5, 6}, {3, 1}, {3.5, 8}, {4, 9}};
+
+    private static int batchSize = 2;
+    private static int nEpochs = 10;
+    private static double learningRate = 1e-4;
+    private static int nClasses = 6;
+    private static List<String> labels;
+
+    private static File modelFilename = new File(System.getProperty("user.dir"),"generated-models/MetalSurfaceDefects_yolov2.zip");
+    private static ComputationGraph model;
+    private static Frame frame = null;
+    public static final Scalar BLUE = RGB(0, 0, 255);
+    public static final Scalar GREEN = RGB(0, 255, 0);
+    public static final Scalar RED = RGB(255, 0, 0);
+    public static final Scalar YELLOW = RGB(255, 225, 0);
+    public static final Scalar PINK = RGB(255, 0, 225);
+    public static final Scalar CYAN = RGB(0, 225, 225);
+    public static Scalar[] colormap = {BLUE,GREEN,RED,YELLOW, PINK, CYAN};
+    private static String labeltext = null;
+
+    public static void main(String[] args) throws Exception {
+
+        MetalDefectDataSetIterator.setup();
+
+        //        STEP 1 : Create iterators
+        RecordReaderDataSetIterator trainIter = MetalDefectDataSetIterator.trainIterator(batchSize);
```

Comment:
dataset cannot be loaded, buggy setup and iterator. please fix

Reply:
verified it's zippyshare problem(similar to gdrive), will switch to another server

Reply:
might not be only the hosting, the paths to the extracted train/test as well. Do another round of self-test once you migrated the file hosting.

Reply:
![image](https://user-images.githubusercontent.com/48118659/73809218-8157a180-480d-11ea-9f23-18b666459261.png)